apps/ios/Shared/Views/Chat/ComposeMessage/ComposeView.swift

```diff
@@ -106,9 +66,52 @@ struct ComposeView: View {
         }
     }
 
-    func loadLinkPreview(_ url: URL) {
-        if url != linkPreview?.uri && url != pendingLinkUrl {
-            pendingLinkUrl = url
+    private func showLinkPreview(_ s: String) {
+        prevLinkUrl = linkUrl
+        linkUrl = parseMessage(s)
+        if let url = linkUrl {
+            if url != linkPreview?.uri && url != pendingLinkUrl {
+                pendingLinkUrl = url
+                if prevLinkUrl == url {
+                    loadLinkPreview(url)
+                } else {
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+                        loadLinkPreview(url)
+                    }
+                }
+            }
+        } else {
+            linkPreview = nil
+        }
+    }
+
+    private func parseMessage(_ msg: String) -> URL? {
+        do {
+            let parsedMsg = try apiParseMarkdown(text: msg)
+            let uri = parsedMsg?.first(where: { ft in
+               ft.format == .uri && !cancelledLinks.contains(ft.text) && !isSimplexLink(ft.text)
+            })
+            if let uri = uri { return URL(string: uri.text) }
+            else { return nil }
+        } catch {
+            logger.error("apiParseMarkdown error: \(error.localizedDescription)")
+            return nil
+        }
+    }
+
+    private func isSimplexLink(_ link: String) -> Bool {
```

Comment:
This is a redefinition of the same function in SendMessageView.kt

Reply:
probably forgot to remove it in one place - I just moved them all together

Reply:
ah - this is in swift